services/httpservice/client.go

```diff
@@ -30,6 +30,38 @@ func IsReservedIP(ip net.IP) bool {
 	return false
 }
 
+func IsOwnIP(ip net.IP) (bool, error) {
+	interfaces, err := net.Interfaces()
+
+	if err != nil {
+		return false, err
+	}
+
+	for _, interf := range interfaces {
+		addresses, err := interf.Addrs()
+
+		if err != nil {
+			return false, err
+		}
+
+		for _, addr := range addresses {
+			var selfIP net.IP
+			switch v := addr.(type) {
+			case *net.IPNet:
```

Comment:
style nit: check both at the same time?
```suggestion
			case *net.IPNet, *net.IPAddr:
```

Reply:
:heavy_check_mark:

Reply:
Didn't know that either, but that's sadly not possible in a type switch:
`The TypeSwitchGuard may include a short variable declaration. When that form is used, the variable is declared at the end of the TypeSwitchCase in the implicit block of each clause. In clauses with a case listing exactly one type, the variable has that type; otherwise, the variable has the type of the expression in the TypeSwitchGuard.`
(from https://golang.org/ref/spec#Type_switches)
Since it will be an `interface{}` that way, the field `IP` doesn't exist. So we will need to stick to 2 clauses for now.

Reply:
Aha! Learned something new myself -- thanks :)